### PR TITLE
Refactor `Disclosure` component to split into 2 (HDS-1932)

### PR DESCRIPTION
**Pull request by KristinLBradley**
### :pushpin: Summary

If merged, this PR will:

- Rename the original `Hds::Disclosure` component to `Hds::MenuPrimitive`
- Add a new `Hds::DisclosurePrimitive` stripped of the “click outside/unfocus/esc to close” functionality

**Previews:**
- MenuPrimitive Showcase: https://hds-showcase-git-hds-1932-refactor-disclosure-hashicorp.vercel.app/utilities/menu-primitive
- MenuPrimitive Web Docs: https://hds-website-git-hds-1932-refactor-disclosure-hashicorp.vercel.app/utilities/menu-primitive
- DisclosurePrimitive Showcase: https://hds-showcase-git-hds-1932-refactor-disclosure-hashicorp.vercel.app/utilities/disclosure-primitive
- DisclosurePrimitive Web Docs: https://hds-website-git-hds-1932-refactor-disclosure-hashicorp.vercel.app/utilities/disclosure-primitive

### :link: External links

Jira ticket: [HDS-XXX](https://hashicorp.atlassian.net/browse/HDS-1932)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Comment by KristinLBradley** [comment]
Is there anything else I should update for either of these "new" components? Anything more I should change in the documentation or anything?

File: packages/components/addon/components/hds/disclosure-primitive/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { schedule } from '@ember/runloop';
+
+export default class HdsDisclosurePrimitiveComponent extends Component {
+  @tracked isOpen; // notice: if in the future we need to add a "@isOpen" prop to control the status from outside (eg to have the DisclosurePrimitive opened on render) just add  "this.args.isOpen" here to initalize the variable
+  @tracked toggleRef;
+
+  @action
+  didInsert(element) {
+    this.element = element;
+  }
```
Comment (didoo) [inline packages/components/addon/components/hds/disclosure-primitive/index.js:18]: I don't think this is needed anymore (it's used in the `onFocusOut` method, but now it's been deleted in this primitive)

```suggestion
```

File: packages/components/addon/components/hds/disclosure-primitive/index.hbs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+{{! template-lint-disable no-invalid-interactive }}
+<div class="hds-disclosure-primitive" ...attributes {{did-insert this.didInsert}}>
```
Comment (didoo) [inline packages/components/addon/components/hds/disclosure-primitive/index.hbs:6]: not needed anymore (see comment below)

```suggestion
<div class="hds-disclosure-primitive" ...attributes>
```

**Review by didoo — APPROVED** [review]
All good for me (I've left a comment re. some dead code that can be removed)

A couple of extra points, up for discussion:
- maybe in the website docs pages, add an info box or a paragraph with cross-links between the `DropdownPrimitive` and the `MenuPrimitive`, so that if someone clicks on one of the doc pages, we tell them there is also the other variant of primitive
- should the `hds-menu-primitive__toggle` be `display: inline-block` to avoid a dead-area next to it when the content is visible (and wider than the "toggle" itself)? any downsides in that case (mainly around layout)?

File: packages/components/app/styles/components/disclosure-primitive.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// MenuPrimitive COMPONENT
+//
+
+.hds-menu-primitive {
```
Comment (alex-ju) [inline packages/components/app/styles/components/disclosure-primitive.scss:10]: blocker: `.hds-disclosure-primitive`

File: packages/components/addon/components/hds/disclosure-primitive/index.hbs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+{{! template-lint-disable no-invalid-interactive }}
```
Comment (alex-ju) [inline packages/components/addon/components/hds/disclosure-primitive/index.hbs:5]: This exception should not be needed anymore

**Review by alex-ju — COMMENTED** [review]
My first impression is that the `DisclosurePrimitive` inherits a bit too much from the `Disclosure`/`MenuPrimitive`. I've left comments where I think we can strip it down even further.

File: packages/components/addon/components/hds/disclosure-primitive/index.hbs (hunk 4 above)
Comment (KristinLBradley) [inline packages/components/addon/components/hds/disclosure-primitive/index.hbs:5]:
```suggestion
```

File: packages/components/addon/components/hds/disclosure-primitive/index.hbs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+<div class="hds-disclosure-primitive" ...attributes {{did-insert this.didInsert}}>
+  <div class="hds-disclosure-primitive__toggle" tabindex="-1">
+    {{yield (hash onClickToggle=this.onClickToggle isOpen=this.isOpen) to="toggle"}}
+  </div>
+  {{#if this.isOpen}}
+    <div class="hds-disclosure-primitive__content" tabindex="-1">
```
Comment (KristinLBradley) [inline packages/components/addon/components/hds/disclosure-primitive/index.hbs:11]:
```suggestion
    <div class="hds-disclosure-primitive__content">
```

File: packages/components/app/styles/components/disclosure-primitive.scss [hunk 6]
```diff
@@ -4,10 +4,10 @@
  */
 
 //
-// DISCLOSURE COMPONENT
+// MenuPrimitive COMPONENT
```
Comment (KristinLBradley) [inline packages/components/app/styles/components/disclosure-primitive.scss:7]:
```suggestion
// DisclosurePrimitive COMPONENT
```

File: packages/components/app/styles/components/disclosure-primitive.scss [hunk 7]
```diff
@@ -4,10 +4,10 @@
...
 //
 
-.hds-disclosure {
+.hds-menu-primitive {
```
Comment (KristinLBradley) [inline packages/components/app/styles/components/disclosure-primitive.scss:10]:
```suggestion
.hds-disclosure-primitive {
```

**Comment by KristinLBradley** [comment]
> All good for me (I've left a comment re. some dead code that can be removed)
> 
> A couple of extra points, up for discussion:
> 
> * maybe in the website docs pages, add an info box or a paragraph with cross-links between the `DropdownPrimitive` and the `MenuPrimitive`, so that if someone clicks on one of the doc pages, we tell them there is also the other variant of primitive
> * should the `hds-menu-primitive__toggle` be `display: inline-block` to avoid a dead-area next to it when the content is visible (and wider than the "toggle" itself)? any downsides in that case (mainly around layout)?

Let me know what you both think, but in my opinion button like elements should be `display: inline-block` to match the default display of native HTML button elements.

@alex-ju Opinions?

**Comment by KristinLBradley** [comment]
I added explanatory text with cross-links in the documentation. Let me know if this looks ok or if there's something more I should do. @didoo @alex-ju 

**See:**
- MenuPrimitive Web Docs: https://hds-website-git-hds-1932-refactor-disclosure-hashicorp.vercel.app/utilities/menu-primitive
- DisclosurePrimitive Web Docs: https://hds-website-git-hds-1932-refactor-disclosure-hashicorp.vercel.app/utilities/disclosure-primitive

**Screenshot with added text highlighted:**
image

File: packages/components/addon/components/hds/disclosure-primitive/index.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { schedule } from '@ember/runloop';
+
+export default class HdsDisclosurePrimitiveComponent extends Component {
+  @tracked isOpen; // notice: if in the future we need to add a "@isOpen" prop to control the status from outside (eg to have the DisclosurePrimitive opened on render) just add  "this.args.isOpen" here to initalize the variable
+  @tracked toggleRef;
```
Comment (alex-ju) [inline packages/components/addon/components/hds/disclosure-primitive/index.js:13]: `toggleRef` is not used anywhere (sorry, somehow this comment didn't made it in the initial review)

File: packages/components/addon/components/hds/disclosure-primitive/index.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+  @action
+  onClickToggle(event) {
+    // we store a reference to the DOM node that has the "onClickToggle" event associated with it
+    if (!this.toggleRef) {
+      this.toggleRef = event.currentTarget;
+    }
```
Comment (alex-ju) [inline packages/components/addon/components/hds/disclosure-primitive/index.js:20]: This logic can go too, there's no need to store the reference on this primitive.

**Review by alex-ju — APPROVED** [review]
Looks good now! One comment on not needing to store a `toggleRef` in the `DisclosurePrimitive`

**Review by didoo — APPROVED** [review]
@KristinLBradley all good to go for me. we can decide if make the toggle an inline block in a different PR (if we need to)

File: .changeset/green-planes-add.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
+---
+"@hashicorp/design-system-components": minor
+"website": minor
```
[the remark this answers is not in the document]
Comment (KristinLBradley) [reply .changeset/green-planes-add.md:3]: Added you as a reviewer: https://github.com/hashicorp/design-system/pull/1374

**Pull request merged**